packages/core/src/routes/session/passwordless.ts

```diff
@@ -148,195 +154,103 @@ export default function passwordlessRoutes<T extends AnonymousRouter>(
     }
   );
 
-  router.post(
-    `${signInRoute}/sms/send-passcode`,
-    koaGuard({ body: object({ phone: string().regex(phoneRegEx) }) }),
-    async (ctx, next) => {
-      const { jti } = await provider.interactionDetails(ctx.req, ctx.res);
-      const { phone } = ctx.guard.body;
-      const type = 'SignInSmsSendPasscode';
-      ctx.log(type, { phone });
+  router.post(`${signInRoute}/sms`, async (ctx, next) => {
+    const { result } = await provider.interactionDetails(ctx.req, ctx.res);
+    const verificationStorage = parseVerificationStorage(result);
 
-      assertThat(
-        await hasUserWithPhone(phone),
-        new RequestError({ code: 'user.phone_not_exists', status: 422 })
-      );
+    const { phone, flow, expiresAt } = verificationStorage;
+    assertThat(phone, new RequestError({ code: 'session.passwordless_not_verified', status: 401 }));
 
-      const passcode = await createPasscode(jti, PasscodeType.SignIn, { phone });
-      const { dbEntry } = await sendPasscode(passcode);
-      ctx.log(type, { connectorId: dbEntry.id });
-      ctx.status = 204;
+    const type = getPasswordlessRelatedLogType(PasscodeType.SignIn, 'sms');
+    ctx.log(type, { phone, flow, expiresAt });
 
-      return next();
-    }
-  );
+    checkVerificationSessionByFlow(PasscodeType.SignIn, verificationStorage);
 
-  router.post(
-    `${signInRoute}/sms/verify-passcode`,
-    koaGuard({ body: object({ phone: string().regex(phoneRegEx), code: string() }) }),
-    async (ctx, next) => {
-      const { jti } = await provider.interactionDetails(ctx.req, ctx.res);
-      const { phone, code } = ctx.guard.body;
-      const type = 'SignInSms';
-      ctx.log(type, { phone, code });
+    assertThat(
+      await hasUserWithPhone(phone),
+      new RequestError({ code: 'user.phone_not_exists', status: 422 })
+    );
+    const { id } = await findUserByPhone(phone);
+    ctx.log(type, { userId: id });
 
-      assertThat(
-        await hasUserWithPhone(phone),
-        new RequestError({ code: 'user.phone_not_exists', status: 422 })
-      );
+    await updateUserById(id, { lastSignInAt: Date.now() });
+    await assignInteractionResults(ctx, provider, { login: { accountId: id } });
 
-      await verifyPasscode(jti, PasscodeType.SignIn, code, { phone });
-      const { id } = await findUserByPhone(phone);
-      ctx.log(type, { userId: id });
+    return next();
+  });
 
-      await updateUserById(id, { lastSignInAt: Date.now() });
-      await assignInteractionResults(ctx, provider, { login: { accountId: id } }, true);
+  router.post(`${signInRoute}/email`, async (ctx, next) => {
+    const { result } = await provider.interactionDetails(ctx.req, ctx.res);
+    const verificationStorage = parseVerificationStorage(result);
```

Comment:
line 183 to line 184 can be wrapped into a function like "getVerficationFromInteraction(ctx.req, ctx.res)"

Reply:
Updated.